### ISSUE TITLE
New Employee Permission and Stock location (with space)

### ISSUE THREAD

**Issue by data2base** [opensourcepos/opensourcepos#2651]
### Background information

__IMPORTANT: If you choose to ignore this issue report template, your issue will be closed as we cannot help without the requested information.__

Please make sure you tick (add an x between the square brackets with no spaces) the following check boxes:

- [] Reporting an issue of an unmodified OSPOS installation
- [] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Installation information

OSPOS Installation Info: | 3.3.0 - 0b9a76
-- | --
en-US
» GD: Enabled ✓» BC Math: Enabled ✓» INTL: Enabled ✓» OpenSSL: Enabled ✓» MBString: Enabled ✓» Curl: Enabled ✓
.Browser: Chrome.Server Software: Apache.PHP Version: 7.0.33.DB Version: 5.7.28-log.Server Port: 80.OS: Linux depro4.fcomet.com 2.6.32-954.3.5.lve1.4.71.el6.x86_64 #1 SMP Fri Sep 6 06:22:16 EDT 2019 x86_64
» [application/logs:] - 0775 \| Writable ✓» [public/uploads:] - 0775 \| Writable ✓» [public/uploads/item_pics:] - 0775 \| Writable ✓» [import_customers.csv:] - 0664 \| Writable ✓

###  Question

Please assist, im creating a new user, but have multiple sale/ stock locations. 
I want to give my employee access to one location, im however not able to do this or dont know the process.

**Comment by daN4cat** [comment]
@data2base under employees, when you select your employee check the permissions tab.
Scroll down and you can give access to specific stock locations in items and sale.

**Comment by data2base** [comment]
Thanks. It gives the option to select the stock locations, but does not save the option?
Please see screenshot. 
![Screenshot_1](https://user-images.githubusercontent.com/37272312/71656144-db62d400-2d42-11ea-8efd-2ff5741071d4.jpg)

**Comment by daN4cat** [comment]
Under Items if you select let say Online Sales and you submit, what happens?

**Comment by data2base** [comment]
It submits but after reload the checkbox is no longer checked.

**Comment by data2base** [comment]
Can i check in the DB? for errors?

**Comment by daN4cat** [comment]
@data2base right, that is weird. It needs some testing. I have just one stock location so I never tested your scenario.

@WebShells @SteveIreland @objecttothis @jekkos @odiea have you experienced anything like the above with multiple stock locations?

**Comment by objecttothis** [comment]
> @data2base right, that is weird. It needs some testing. I have just one stock location so I never tested your scenario.
> 
> @WebShells @SteveIreland @objecttothis @jekkos @odiea have you experienced anything like the above with multiple stock locations?

I too have only one stock location but will eventually move to multiple. If you enable the debug log level to include debug messages it should show you if there is an error saving the value.

**Comment by WebShells** [comment]
Is this a new installation or a migrated db? Personally i was able to migrate to 3.2 and permissions were working without any problems on several stock locations, will test it on the last stable release in few and let you know.

**Comment by jekkos** [comment]
Just guessing, but maybe it's related to spaces in the stock location name?

**Comment by odiea** [comment]
I did not until just now. Here is the result. I tried the name with and without spaces with no issues.

![2020-01-02 14_17_43-Window](https://user-images.githubusercontent.com/10962177/71693911-e7628c00-2d6a-11ea-95a8-5e6a21ac9d5c.png)

**Comment by objecttothis** [comment]
It seems that we need to either not allow spaces or properly process stock location names that have spaces. Most admins are you to want spaces in their stock location names since their users will better understand stock locations names that way. When we migrate to multiple stock locations we won't have spaces because our stock locations will correlate to a location code we use internally.

**Comment by data2base** [comment]
oh great, thanks, can i just change the stock location names, it wont have an effect on any data?

**Comment by daN4cat** [comment]
@data2base it will not affect your data.

**Comment by daN4cat** [comment]
Team, I'm not sure I get if in the end the space in the stock location is an issue when setting the permission or not. Can somebody clearly confirm? Thanks in advance.

**Comment by jekkos** [comment]
I just did a quick test and
* using spaces in a name seems to be fine, location shows up in product and permission is created
* renaming a location also renames the permission for each employee

**Comment by daN4cat** [comment]
@data2base not sure why you get that issue. Have you fixed it renaming the stock location?

**Comment by data2base** [comment]
@daN4cat Changed mine and working now. @jekkos did a test.

**Comment by WebShells** [comment]
@daN4cat no issues actually in the assigning user permissions with or without spaces in stockname, I remember spaces in stock names were managed in previous PRs... To what i remember we already covered something related to spaces issue...
@data2base can you try migrating to our latest stable release ?
